mesa/experimental/devs/simulator.py

```diff
@@ -0,0 +1,265 @@
+import numbers
+from typing import Any, Callable
+
+from .eventlist import EventList, Priority, SimulationEvent
+
+
+class Simulator:
+    """The Simulator controls the time advancement of the model.
+
+    The simulator uses next event time progression to advance the simulation time, and execute the next event
+
+    Attributes:
+        event_list (EventList): The list of events to execute
+        time (float | int): The current simulation time
+        time_unit (type) : The unit of the simulation time
+        model (Model): The model to simulate
+
+
+    """
+
+    # TODO: add replication support
+    # TODO: add experimentation support
+
+    def __init__(self, time_unit: type, start_time: int | float):
+        # should model run in a separate thread,
+        # and we can then interact with start, stop, run_until, and step?
+        self.event_list = EventList()
+        self.time = start_time
```

Comment:
In #1942 we added a clock to the model, in `self._steps` and `self._time`. How do you - conceptually - see this integrating on the long term? i.e. can there be one scheduler that commands multiple models, and is then the scheduler time leading? Is there one model with one scheduler? If so, which is the "atomic clock" so to speak.

Reply:
In the long term, this should replace `self._steps` and `self._time` in `Model`. The simulator would become the sole truth for time.

Reply:
- Right, in that case, can we still https://github.com/projectmesa/mesa/discussions/1912?

Reply:
Yes. It will probably make it even easier.